### DISCUSSION TITLE
Problemas com Teste em Testando as Entidades e VOs

### DISCUSSION THREAD

**Discussion by raphaelbonfim** [balta-io/forum#503] [Tira Dúvidas] [answered]
Boa tarde, estou com dificuldades pra ajustar os testes da aula "Testando as Entidades e VOs" em "Modelando Domínios Ricos".

Meu problema é o seguinte:
![image](https://github.com/balta-io/forum/assets/61986203/b06aec12-1d87-4af2-897c-4363fa34c525)

a subscription chega por parâmetro já com o pagamento adicionado, mas ela não aparece no foreach, o que acaba por nao ativar a assinatura

![image](https://github.com/balta-io/forum/assets/61986203/eb7ead94-96eb-49ae-ac22-249813475918)

o teste foi feito para dar erro ao tentar adicionar uma assinatura em uma conta com uma assinatura já feita.

Alguem pode me ajudar? quero corrigir os testes que foram feitos na aula e deixar tudo redondo

**Comment by andrebaltieri** [comment, marked as answer]
Boa noite, @raphaelbonfim, como vai?
Pelo que me lembro, este método verifica se o aluno já tem uma assinatura ativa e causa uma notificação caso isto seja verdade.

Seu problema seria popular as assinaturas do aluno? Se sim, acredito que isto é feito em outro método.

**Reply by raphaelbonfim** [reply]
Boa noite professor, eu dei uma olhada no código fonte da aula e encontrei o método que fica no final "AddSubscription", mas ele tem uma condição em que o "Valid" do "Notification" da false, ele ele pula o método que popula as subscriptions.

![image](https://github.com/balta-io/forum/assets/61986203/c43a005e-4cfc-41de-beee-9489d785bfba)

como eu corrijo? eu nego esse "Valid"? ou estou tampando o sol com a peneira assim?

**Reply by andrebaltieri** [reply]
@raphaelbonfim checou se as subscriptions estão tentando ser carregadas pelo `Include` então? Esta seria outra forma de se fazer isto.
Em adicional, existem assinaturas vinculadas ao aluno que está testando no banco? Senão ele não preenche mesmo.

**Reply by raphaelbonfim** [reply]
É no curso de "Modelando domínios ricos" nas aulas:  "Design By Contracts" e "Testando as Entidades e VOs"

No curso não está conectado ao banco ainda, estamos fazendo os testes pros métodos, na aula mesmo da erro nos testes e você passa pra próxima aula pra não atrasar o curso, ai eu to tentando resolver, mas to um pouco confuso em como relaciona uma subscription pro aluno. No caso estamos fazendo um aluno fictício no teste, e passando ele pro método, mas a sub nunca é preenchida, ai eu me perdi. 

se puder dar uma olhada e ver se ta certo mesmo, eu agradeceria.

**Reply by andrebaltieri** [reply]
Bom dia, @raphaelbonfim, entendi! Acho que não fizemos o método para adicionar assinatura ao aluno no curso.

Segue:
```csharp
public void AddSubscription(Subscription subscription)
{
    _subscriptions.Add(subscription);
}
```

Agora ao chamar o método `AddSubscription` você terá uma assinatura atribuída ao aluno.

**Reply by raphaelbonfim** [reply]
Vlw professor!
